app/src/main/java/waiter/Awaiter.java

```diff
@@ -0,0 +1,7 @@
+package waiter;
+
+import java.io.IOException;
+
+public interface Awaiter {
+    Connection awaitClient() throws IOException;
```

Comment:
Do we need this interface?

Reply:
I included this interface so I can easily mock the Listener without having to override the Listener class